### PR TITLE
Fix/Handle subnet removal

### LINKED ISSUE #1162: Remove subnet from netmap candidates on subnet remove event
https://github.com/nspcc-dev/neofs-node/blob/1667ec9e6dc7f292365719f203f48282d8f9e076/pkg/innerring/subnet.go#L315

### PR DESCRIPTION
**Pull request by carpawell**
Closes #1162.

### CODE REVIEW

**Comment by alexvanin** [comment]
Not blocked anymore, neofs-dev-env uses neofs-contract v0.15.0 release with the fix.

File: pkg/innerring/subnet.go [hunk 1]
```diff
@@ -315,5 +319,93 @@ func (s *Server) handleSubnetRemoval(e event.Event) {
...
+	if err != nil {
+		s.log.Error("getting netmap candidates",
+			zap.Error(err),
+		)
+
+		return
+	}
+
+	var removedID subnetid.ID
+	_ = removedID.Unmarshal(delEv.ID()) // parsing errors were checked above
```
Comment (cthulhu-rider) [inline pkg/innerring/subnet.go:334]: Too risky IMO. We need to get `removeID` from the place where it was checked `above`, or check for errors explicitly.

File: pkg/innerring/subnet.go [hunk 2]
```diff
@@ -315,5 +319,93 @@ func (s *Server) handleSubnetRemoval(e event.Event) {
...
+		}
+
+		// not remove from netmap if there
+		// is at least one more subnet
+		removeFromNM = false
+
+		return nil
+	})
+	if err != nil {
+		s.log.Error("iterating node's subnets",
```
Comment (cthulhu-rider) [inline pkg/innerring/subnet.go:368]: `s.log` expected, maybe u wanted `log`?

File: pkg/innerring/subnet.go [hunk 3]
```diff
@@ -315,5 +319,93 @@ func (s *Server) handleSubnetRemoval(e event.Event) {
...
+			zap.Error(err),
+		)
+	}
+
+	// remove from netmap if there are no
+	// more subnets for the node or if it
+	// is impossible to process its subnet
+	// information
+	if err != nil || removeFromNM {
```
Comment (cthulhu-rider) [inline pkg/innerring/subnet.go:377]: Pay attention on `IterateSubnets` docs: it returns an error if node has no or has been removed from all subnets. So we can get rid of `removeFromNM`.

Note that in this case this `if` statement and the one above can be merged.

Comment (carpawell) [reply pkg/innerring/subnet.go:377]: yes, read the doc but missed "...not included in any subnet **by the end of the loop**", thanks, fixed

File: pkg/innerring/subnet.go (hunk 1 above)
Comment (carpawell) [reply pkg/innerring/subnet.go:334]: ok, agree

File: pkg/innerring/subnet.go (hunk 3 above)
Comment (carpawell) [reply pkg/innerring/subnet.go]: oh no: i remembered why i did so: yes, i read the doc correctly but did not want to merge two cases (printing the same type of logs and so on): "could not read the subnet ID" (which seems an error) and "node does not have any more subnet " (which seems as ok situation)

on the other hand that is not that critical so maybe leave them merged, what do you think?

Comment (cthulhu-rider) [reply pkg/innerring/subnet.go]: `node does not have any more subnet` is expected to raise pretty rare, so I'd print it anyway.

**Review by cthulhu-rider — APPROVED** [review]

File: pkg/innerring/subnet.go (hunk 3 above)
Comment (carpawell) [reply pkg/innerring/subnet.go]: @cthulhu-rider so maybe we need to have public `ErrNoSubnet` in SDK

**Pull request merged**